homeassistant/components/lcn/__init__.py

```diff
@@ -147,26 +159,26 @@ def unique_id(self):
         return f"{self.entry_id}-{unique_device_id}-{self.config[CONF_RESOURCE]}"
 
     @property
-    def should_poll(self):
+    def should_poll(self) -> bool:
         """Lcn device entity pushes its state to HA."""
         return False
 
-    async def async_added_to_hass(self):
+    async def async_added_to_hass(self) -> None:
         """Run when entity about to be added to hass."""
         if not self.device_connection.is_group:
             self._unregister_for_inputs = self.device_connection.register_for_inputs(
                 self.input_received
             )
 
-    async def async_will_remove_from_hass(self):
+    async def async_will_remove_from_hass(self) -> None:
         """Run when entity will be removed from hass."""
         if self._unregister_for_inputs is not None:
-            self._unregister_for_inputs()
+            self._unregister_for_inputs()  # type: ignore[unreachable]
```

Comment:
Instead of this, add a type hint to `self._unregister_for_inputs` in the constructor